storage/volume/volume.go

```diff
@@ -0,0 +1,166 @@
+// Copyright 2015 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package volume
+
+import (
+	"github.com/juju/errors"
+	"github.com/juju/names"
+
+	"github.com/juju/juju/state"
+)
+
+// NewVolumeManager returns a VolumeManager implementation using the specified state.
+func NewVolumeManager(st VolumeState) VolumeManager {
+	return &volumeManager{vst: st}
+}
+
+var _ VolumeManager = (*volumeManager)(nil)
+
+type volumeManager struct {
+	vst VolumeState
+}
+
+type VolumeState interface {
+	AllBlockDevices() ([]state.BlockDevice, error)
+}
+
+// List is defined on VolumeManager interface.
+func (pm *volumeManager) List() ([]Disk, error) {
+	devices, err := pm.vst.AllBlockDevices()
+	if err != nil {
+		return nil, errors.Annotate(err, "listing block devices")
+	}
+	attachments := make([]Attachment, len(devices))
+	for i, d := range devices {
+		attachments[i] = pm.constructAttachment(d)
+	}
+	// TODO(anastasiamac 2015-01-30) since volumes don't really exist yet,
+	// only return one disk for now
+	volumes := make([]Disk, 1)
```

Comment:
Currently, there is a one-to-one relation between BlockDevice (which doubles as VolumeAttachment) and Volume.